pyiron_gui/project/project_browser.py

```diff
@@ -23,6 +29,207 @@
+
+    def __repr__(self):
+        return repr(self._wrapped_object)
+
+    def self_representation(self):
+        """Self representation of the wrapped object if known, else None is returned."""
+        if self._type is None:
+            return
+        if self._type == "structure":
+            return self._wrapped_object.plot3d()
+        if self._type == 'murnaghan':
+            plt.ioff()
+            self._wrapped_object.plot()
+
+
+class DisplayOutputGUI:
+
+    """Display various kind of data in an appealing way using a ipywidgets.Output inside an ipywidgets.Vbox
+    The behavior is very similar to standard ipywidgets.Output except one has to pass cls.box to get a display."""
+    def __init__(self, *args, **kwargs):
+        self.box = widgets.VBox(*args, **kwargs)
+        self.buttons = widgets.HBox()
+        self.output = widgets.Output(layout=widgets.Layout(width='100%'))
+        self.fig = None
+        self.ax = None
+        self._debug = False
+        self.refresh()
+
+    def refresh(self):
+        self.box.children = (self.buttons, self.output)
+
+    def __enter__(self):
+        """Use context manager on the widgets.Output widget"""
+        self.buttons = widgets.HBox()
+        return self.output.__enter__()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Use context manager on the widgets.Output widget"""
+        self.output.__exit__(exc_type, exc_val, exc_tb)
+
+    def __getattr__(self, item):
+        """Forward unknown attributes to the widgets.Output widget"""
+        return self.output.__getattribute__(item)
+
+    def clear_output(self, *args, **kwargs):
+        clear_button = kwargs.pop('clear_button', False)
+        if clear_button:
+            self.buttons = widgets.HBox()
+        self.output.clear_output(*args, **kwargs)
+        self.refresh()
+
+    def _update_buttons(self, obj):
+
+        def click_button(b):
+            self.output.clear_output()
+            self.display(obj=obj)
+
+        if isinstance(obj, PyironWrapper) and obj.has_self_representation:
+            button = widgets.Button(description="Re-plot " + obj.name)
+            button.on_click(click_button)
+            self.buttons.children = tuple([button])
+
+        self.refresh()
+
+    def display(self, obj, default_output=None):
+        self._update_buttons(obj)
+        with self.output:
+            if obj is None and default_output is None:
+                raise TypeError("Given 'obj' is of 'NoneType'.")
+            elif obj is None:
+                print(default_output)
+            elif isinstance(obj, PyironWrapper):
+                plt.ioff()
+                to_display = obj.self_representation()
+                if to_display is not None:
+                    display(obj.self_representation())
+            else:
+                plt.ioff()
+                display(self._output_conv(obj))
+
+    def _output_conv(self, obj):
+        if hasattr(obj, '_repr_html_'):
```

Comment:
Readability nit) Could we slide the `eol` definition and debug print up and the `hasattr(obj, '_repr_html_') down, and change the `if` in l181 to an `elif`? Then if have a contiguous `if/elif/.../elif/else` block each clause of which contains a `return` call.